cluster-autoscaler/cloudprovider/gce/autoscaling_gce_client_test.go

```diff
@@ -110,32 +109,41 @@ func TestWaitForOpError(t *testing.T) {
 	defer server.Close()
 	g := newTestAutoscalingGceClient(t, "project1", server.URL, "")
 
-	server.On("handle", "/projects/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197").Return(operationDoneResponseError).Once()
-
-	operation := &gce_api.Operation{Name: "operation-1505728466148-d16f5197"}
+	server.On("handle", "/projects/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197/wait").Return(operationDoneResponseError).Once()
 
-	err := g.waitForOp(operation, projectId, zoneB, false)
+	err := g.WaitForOperation("operation-1505728466148-d16f5197", "TestWaitForOpError", projectId, zoneB)
 	assert.Error(t, err)
+	mock.AssertExpectationsForObjects(t, server)
 }
 
 func TestWaitForOpTimeout(t *testing.T) {
 	server := test_util.NewHttpServerMock()
 	defer server.Close()
 	g := newTestAutoscalingGceClient(t, "project1", server.URL, "")
 
-	// The values here are higher than in other tests since we're aiming for timeout.
-	// Lower values make this fragile and flakey.
-	g.operationPollInterval = 10 * time.Millisecond
-	g.operationWaitTimeout = 49 * time.Millisecond
-
-	// Sometimes, only 3 calls are made, but it doesn't really matter,
-	// so let's not assert expectations for this mock, just check for timeout error.
-	server.On("handle", "/projects/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197").Return(operationRunningResponse).Times(5)
+	// default polling interval and wait time are too big for the test
+	g.operationWaitTimeout = 1 * time.Millisecond
+	g.operationPollInterval = 1 * time.Millisecond
 
-	operation := &gce_api.Operation{Name: "operation-1505728466148-d16f5197"}
+	server.On("handle", "/projects/project1/zones/us-central1-b/operations/operation-1505728466148-d16f5197/wait").Return(operationRunningResponse).Once()
 
-	err := g.waitForOp(operation, projectId, zoneB, false)
+	err := g.WaitForOperation("operation-1505728466148-d16f5197", "TestWaitForOpTimeout", projectId, zoneB)
 	assert.Error(t, err)
+	mock.AssertExpectationsForObjects(t, server)
+}
+
+func TestWaitForOpContextTimeout(t *testing.T) {
+	server := test_util.NewHttpServerMock()
+	defer server.Close()
+	g := newTestAutoscalingGceClient(t, "project1", server.URL, "")
+
+	g.operationWaitTimeout = 1 * time.Millisecond
```

Comment:
I'd make this a bit higher, 1ms probably means that the context is cancelled immediately, before even reaching the API call (and increase the value in `After` below proportionally).

Reply:
Agree, increased timeout to 10 ms and After to 1 minute in order to better reflect that we are testing context timeout during API call